### PR TITLE
feat: drift and proxy commands article

### PR DESCRIPTION
**Pull request by sobanieca-redocly**
## What/Why/How?

Added article about `drift` and `proxy` commands.

## Check yourself

- [ ] Code is linted
- [ ] Tested
- [ ] All new/updated code is covered with tests

## Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### CODE REVIEW

File: blog/metadata/blog-metadata.yaml [hunk 1]
```diff
@@ -55,6 +55,10 @@ authors:
     name: Vitalii Skulskyi
     authorBIO: Software Engineer, Redocly
     image: skulskyi.jpg
+  - id: adam-sobaniec
+    name: Adam Sobaniec
+    authorBIO: Software Engineer, Redocly
+    image: sobaniec.jpg
```
Comment (sobanieca-redocly) [inline blog/metadata/blog-metadata.yaml]: I don't know how to convert my image to the format that we usually use. Need some help here.

File: blog/catch-api-drift.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  - redocly:product-updates
+  - api-testing:integration-testing
+---
+
+Your OpenAPI description says one thing.
+Your API does another.
+Nobody notices, because nothing fails — until an SDK generated from that description breaks in a customer's build, or a partner integration chokes on a field that was "documented" but never returned.
+
+That gap between what you describe and what you ship has a name: drift.
+And it accumulates quietly, one hotfix and one forgotten field at a time.
```
Comment (JLekawa) [inline blog/catch-api-drift.md]:
```suggestion
The gap between what you describe and what you ship has a name: drift.
It accumulates quietly, one hotfix and one forgotten field at a time.
```

File: blog/catch-api-drift.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+Redocly CLI now ships two commands that work together to close that gap: `proxy` records what your API actually does, and `drift` compares it against what your OpenAPI description claims.
```
Comment (JLekawa) [inline blog/catch-api-drift.md]:
```suggestion
Redocly CLI ships two commands that work together to close this gap: `proxy` records what your API actually does, and `drift` compares it against what your OpenAPI description claims.
```

File: blog/catch-api-drift.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  ↳ sample exchange=0 GET /menu (200) listMenuItems
+    expected: type "string"
+    actual: 5
+
+▲ WARN → Undocumented query parameter in traffic: "category"
+  ↳ sample exchange=1 GET /menu (200) listMenuItems
+```
+
+The error is the discrepancy we planted: the API returns a number where the description now promises a string.
+The warning is a bonus we didn't plant at all — the `category` query parameter we sent isn't documented anywhere in the Cafe description.
```
Comment (JLekawa) [inline blog/catch-api-drift.md]:
```suggestion
The warning is a bonus we didn't plant at all. 
The `category` query parameter we sent isn't documented anywhere in the Cafe description.
```

File: blog/catch-api-drift.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+Two requests of traffic, and drift already surfaced something real.
+
+## More than schemas
+
+Findings come from built-in rules that you can select with the `--rules` flag:
+
+- `undocumented-endpoint` flags traffic that doesn't match any documented operation.
+- `schema-consistency` validates parameters, headers, and request/response bodies against your schemas.
+- `security-baseline` checks that requests actually satisfy the security requirements your description declares, and flags things like credentials sent over plain HTTP.
+- `owasp-api-top10` is an opt-in set of heuristics based on the OWASP API Security Top 10 — enable it with `--rules owasp-api-top10` to scan recorded traffic for common API risks.
```
Comment (JLekawa) [inline blog/catch-api-drift.md]:
```suggestion
- `owasp-api-top10` scans recorded traffic for common API risks with heuristics based on the OWASP API Security Top 10 — enable it with `--rules owasp-api-top10`.
```

**Review by JLekawa — APPROVED** [review]

**Comment by sobanieca-redocly** [comment]
> I don't have permissions to push changes to this PR so need to wait to get some to fix pointed issues.

SOLVED.

File: blog/catch-api-drift.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+Here is the setup we like most: you probably already have e2e or integration tests that exercise your API.
+That traffic is a free, realistic sample of how your API actually behaves — you're just letting it evaporate after every run.
+
+Instead, start `proxy` before the test run, route the tests through it, and stop it when they finish:
+
+```bash
+redocly proxy --target http://localhost:9000 --har ./test-traffic.har &
+PROXY_PID=$!
+# run your e2e or integration tests against http://localhost:4040
+kill "$PROXY_PID" && wait "$PROXY_PID"
```
[the remark this answers is not in the document]
Comment (sobanieca-redocly) [reply blog/catch-api-drift.md:67]: Let's not make it too complicated. This is supposed to just show general idea, not super clear script that handles all edge cases

File: blog/catch-api-drift.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+Instead, start `proxy` before the test run, route the tests through it, and stop it when they finish:
+
+```bash
+redocly proxy --target http://localhost:9000 --har ./test-traffic.har &
+# run your e2e or integration tests against http://127.0.0.1:4040
+redocly drift ./test-traffic.har --api ./openapi.yaml
+```
+
+The `drift` command exits with a non-zero code when it finds error-level problems, so it slots into CI as a gate.
+Every test run now doubles as a contract check, with zero extra test code to maintain.
```
[the remark this answers is not in the document]
Comment (sobanieca-redocly) [reply blog/catch-api-drift.md:74]: In CI HAR contains artificial traffic, we should be safe here. There is a not added about HAR files containing sensitive data already in the article.

File: blog/catch-api-drift.md [hunk 8]
```diff
@@ -0,0 +1,137 @@
+---
+template: ../@theme/templates/BlogPost
+title: Catch API drift with the new proxy and drift commands
+description: Record real HTTP traffic with the proxy command, then let the drift command tell you where your API and your OpenAPI description disagree.
+seo:
+  title: Catch API drift with the new proxy and drift commands
+  description: Record real HTTP traffic with the proxy command, then let the drift command tell you where your API and your OpenAPI description disagree.
+author: adam-sobaniec
+publishedDate: "2026-07-08"
```
[the remark this answers is not in the document]
Comment (sobanieca-redocly) [reply blog/catch-api-drift.md:9]: This is no longer valid. Also I'm not sure if this is valid comment. User may want to create PR with some blog article with specific publish date when he plans to merge PR.

File: blog/catch-api-drift.md [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+- `schema-consistency` validates parameters, headers, and request/response bodies against your schemas.
+- `security-baseline` checks that requests actually satisfy the security requirements your description declares, and flags things like credentials sent over plain HTTP.
+  Loopback hosts such as `localhost` and `127.0.0.1` are exempt from the transport check, so sandboxed recordings against a local target stay warning-free.
+- `owasp-api-top10` scans recorded traffic with heuristics based on the OWASP API Security Top 10 - enable it with `--rules owasp-api-top10`. It looks for credential-like query parameters, insecure CORS (wildcard origin with credentials enabled), weak cookie attributes, sensitive-looking fields in response payloads, and large unpaginated responses.
+
+One caveat before you wire `owasp-api-top10` into a CI gate: it inspects each exchange in isolation.
+It has no notion of user identity or ownership across requests, so it cannot detect authorization logic flaws like Broken Object Level Authorization.
+Treat it as an extra pair of eyes on traffic you already have - not as a replacement for security testing.
```
Comment (RomanHotsiy) [inline blog/catch-api-drift.md]: I think this is clear anyways, i would not overload readers with this

**Review by RomanHotsiy — COMMENTED** [review]
This is great. Left one minor comment.

I also think that adding some image or mermaid diagram describing the proxy setup would make it better.

File: blog/catch-api-drift.md (hunk 9 above)
Comment (sobanieca-redocly) [reply blog/catch-api-drift.md]: Removed. Thanks.
